### PR TITLE
Books can accept ratings and stored ratings in DB

### PR DESCRIPTION
**Pull request by louismacvux**
#129 
#130 
Books will compute average ratings and stored updated data in DB whenever a review is posted.

### CODE REVIEW

File: bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/controller/BookController.java [hunk 1]
```diff
@@ -125,7 +125,7 @@
             if(foundBook!=null){
                 //Update and save book
                 foundBook.setDescription(bookUpdateInput.getDesc());
-                foundBook.setGenre(bookUpdateInput.getGenre());
```
Comment (LukeBMorrow) [inline bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/controller/BookController.java:128]: Why was this removed?

File: bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java [hunk 2]
```diff
@@ -16,7 +18,8 @@
     private String description;
     private String author;
     private String isbn;
-    private String genre;
```
Comment (LukeBMorrow) [inline bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java:19]: Same as above, genre is necessary

File: bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/input/BookUpdateInput.java [hunk 3]
```diff
@@ -1,10 +1,13 @@
 package com.bookwyrm.backend.book.input;
 
+import java.util.Map;
+import java.util.HashMap;
+
 public class BookUpdateInput {
 
     private String desc;
     private String id;
-    private String genre;
```
Comment (LukeBMorrow) [inline bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/input/BookUpdateInput.java:7]: Here as well.

File: bookwyrm_backend/src/test/java/com/bookwyrm/backend/unit/book/controller/BookControllerTests.java [hunk 4]
```diff
@@ -134,7 +134,8 @@ public void testGoodUpdate() {
         BookUpdateInput bookUpdateInput = new BookUpdateInput();
         bookUpdateInput.setId("testId");
         bookUpdateInput.setDesc("testDescUpdate");
-        bookUpdateInput.setGenre("testGenre");
```
Comment (LukeBMorrow) [inline bookwyrm_backend/src/test/java/com/bookwyrm/backend/unit/book/controller/BookControllerTests.java:137]: I'm largely just noting where the genre needs to be added back.

File: bookwyrm_backend/src/main/java/com/bookwyrm/backend/review/controller/ReviewController.java [hunk 5]
```diff
@@ -13,6 +15,7 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.awt.print.Book;
```
Comment (LukeBMorrow) [inline bookwyrm_backend/src/main/java/com/bookwyrm/backend/review/controller/ReviewController.java:18]: Is this being used?

File: bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java [hunk 6]
```diff
@@ -16,7 +18,8 @@
...
+    private Map<String, Float> avg;
+    private Map<String, Integer> num;
```
Comment (LukeBMorrow) [inline bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java:22]: num could use a better name. what is it the number of? if it's ratings, then perhaps `ratingNum` would be more appropriate. avg should probably also be ratingAvg

**Review by LukeBMorrow — CHANGES_REQUESTED** [review]
Nearly there, just undoing some deletions of other features.
And two variables should have better names

**Comment by louismacvux** [comment]
My mistake. If that's the official genre for the book, I will add it back

File: bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java (hunk 2 above)
Comment (louismacvux) [reply bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java:19]: My mistake. if that's the official genre for the book, I'll add it back.

Comment (LukeBMorrow) [reply bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/dao/BookDao.java:19]: yes, thats the official genre

**Review by LukeBMorrow — APPROVED** [review]
Looks good!
